src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -544,6 +488,75 @@ void CodeGen::genHWIntrinsic_R_R_RM_I(GenTreeHWIntrinsic* node, instruction ins)
     }
 }
 
+// genHWIntrinsicJumpTableFallback : generate the jump-table fallback for imm-intrinsics
+//                       with non-constant argument
+//
+// Arguments:
+//    intrinsic      - intrinsic ID
+//    nonConstImmReg - the register contains non-constant imm8 argument
+//    baseReg        - a register for the start of the switch table
+//    offsReg        - a register for the offset into the switch table
+//    emitSwCase     - the lambda to generate siwtch-case
+//
+// Return Value:
+//    generate the jump-table fallback for imm-intrinsics with non-constant argument.
+// Note:
+//    This function can be used for all imm-intrinsics (whether full-range or not),
+//    The compiler front-end (i.e. importer) is responsible to insert a range-check IR
+//    (GT_HW_INTRINSIC_CHK) for imm8 argument, so this function does not need to do range-check.
+//
+template <typename HWIntrinsicSwitchCaseBody>
+void CodeGen::genHWIntrinsicJumpTableFallback(NamedIntrinsic            intrinsic,
+                                              regNumber                 nonConstImmReg,
+                                              regNumber                 baseReg,
+                                              regNumber                 offsReg,
+                                              HWIntrinsicSwitchCaseBody emitSwCase)
+{
+    assert(nonConstImmReg != REG_NA);
+    emitter* emit = getEmitter();
+
+    const unsigned maxByte = (unsigned)Compiler::immUpperBoundOfHWIntrinsic(intrinsic) + 1;
+    assert(maxByte <= 256);
+    BasicBlock* jmpTable[256];
```

Comment:
nit: `BasicBlock* jmpTable[maxByte]`?

Reply:
`maxByte` is not a compile-time constant.

Reply:
right, missed that 😄